### PR TITLE
Example, move rbtree to cc and update rbtree

### PR DESCRIPTION
**Pull request by ganwell**
@winged please review all commits at once

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -69,9 +69,9 @@
 master_doc = 'index'
 
 # General information about the project.
-project = 'Chirp'
+project = 'libchirp'
 copyright = (
-    '%s, Jean-Louis Fuchs, Sven Osterwalder, '
+    '%s, Jean-Louis Fuchs, Sven Osterwalder, David Vogt'
```
Comment (winged) [inline doc/conf.py]: Whitespace, comma missing, otherwise it's going to be "David VogtAdfinis"

File: doc/index.rst [hunk 2]
```diff
@@ -1,18 +1,171 @@
...
+  send a completed-message and wait for it. If you overload your peer you will
+  get timeout errors. It also means that chirp is not yet routing friendly.
+
+* Retry: There is a very rare edge-case: if a connections is
+  garbage-collected just when the peer sends a message, you will get a
+  disconnected error. With retry these errors will not appear.
+
+* Load is not reported
+
+We want to keep the interfaces, so the versions implementing these feature won't
```
Comment (winged) [inline doc/index.rst]: "these feature" -> should be plural

**Comment by ganwell** [comment]
@winged b6ab3ad fixes your comments.

File: README.rst [hunk 3]
```diff
@@ -81,7 +71,7 @@ Consequences
 
 * Load is not reported
 
-We want to keep the interfaces, so the versions implementing these feature won't
+We want to keep the interfaces, so the versions implementing theses feature won't
```
Comment (winged) [inline README.rst]: erm.. "these features", not "theses feature" :p

Comment (ganwell) [reply README.rst]: FOoo

**Comment by ganwell** [comment]
@winged foo.... amended it.

**Pull request merged**